src/main/java/gregtech/api/capability/impl/BoilerRecipeLogic.java

```diff
@@ -0,0 +1,282 @@
+package gregtech.api.capability.impl;
+
+import gregtech.api.capability.IMultipleTankHandler;
+import gregtech.api.recipes.ModHandler;
+import gregtech.api.util.GTLog;
+import gregtech.common.ConfigHolder;
+import gregtech.common.metatileentities.multi.MetaTileEntityLargeBoiler;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import java.util.Collections;
+
+import static gregtech.api.capability.GregtechDataCodes.BOILER_HEAT;
+import static gregtech.api.capability.GregtechDataCodes.BOILER_LAST_TICK_STEAM;
+
+// TODO Move to common
+
+public class BoilerRecipeLogic extends AbstractRecipeLogic {
+
+    private static final long STEAM_PER_WATER = 160;
+
+    private int currentHeat;
+    private int lastTickSteamOutput;
+    private int excessWater, excessFuel, excessProjectedEU;
+
+    public BoilerRecipeLogic(MetaTileEntityLargeBoiler tileEntity) {
+        super(tileEntity, null);
+    }
+
+    @Override
+    public void update() {
+        if (!isActive() && currentHeat > 0) {
```

Comment:
Should we also lose heat when the boiler is paused with a machine controller or soft hammer?